packages/integration-tests/suites/tracing/baggage/test.ts

```diff
@@ -9,8 +9,8 @@ sentryTest('should send baggage data in transaction envelope header', async ({ g
 
   const envHeader = await getFirstSentryEnvelopeRequest<EventEnvelopeHeaders>(page, url, envelopeHeaderRequestParser);
 
-  expect(envHeader.baggage).toBeDefined();
-  expect(envHeader.baggage).toEqual(
+  expect(envHeader.trace).toBeDefined();
+  expect(envHeader.trace).toEqual(
```

Comment:
The envelope header is independent of baggage or tracestate encoding. It is directly a JSON object with all the keys relevant to Sentry. See: https://develop.sentry.dev/sdk/performance/trace-context/#envelope-headers

Instead of serializing baggage, please filter for `sentry-*` keys and directly add them as object to the header.